File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/features/anchorboardcast/AnchorViewDefine.java
package com.trtc.uikit.livekit.features.anchorboardcast;

public class AnchorViewDefine {

    public static class AnchorState {
        public long viewCount;
        public long duration;
        public long messageCount;
        public long giftIncome;
        public long giftSenderCount;
        public long likeCount;
    }

    public interface AnchorViewListener {
        void onEndLiving(AnchorState state);

        void onClickFloatWindow();
    }

    public enum RoomBehavior {
        CREATE_ROOM,
        ENTER_ROOM;
    }
}

File: iOS/TUILiveKit/Sources/Features/AnchorBoardcast/AnchorViewDefine.swift
//
//  AnchorViewDefine.swift
//  TUILiveKit
//
//  Created by gg on 2025/6/20.
//

import Foundation

public class AnchorState {
    public var duration: Int = 0
    public var viewCount: Int = 0
    public var messageCount: Int = 0
    public var giftTotalCoins: Int = 0
    public var giftTotalUniqueSender: Int = 0
    public var likeTotalUniqueSender: Int = 0
}

public protocol AnchorViewDelegate: AnyObject {
    func onClickFloatWindow()
    func onEndLiving(state: AnchorState)
}

public enum RoomBehavior {
    case createRoom
    case enterRoom
}
